pkg/providers/pricing/pricing.go

```diff
@@ -201,7 +167,7 @@ func (p *Provider) SpotPrice(instanceType string, zone string) (float64, bool) {
 	return 0.0, false
 }
 
-func (p *Provider) updatePricing(ctx context.Context) {
+func (p *Provider) UpdatePricing(ctx context.Context) {
```

Comment:
Maybe move this func into the controller portion? Also, can we do a `workqueue.ParallelizeUntil` to abstract away waitGroup logic?